app/models/membership_application.rb

```diff
@@ -36,6 +36,7 @@ class MembershipApplication < ApplicationRecord
   accepts_nested_attributes_for :uploaded_files, allow_destroy: true
 
   scope :open, -> { where.not(state: [:accepted, :rejected]) }
+  scope :current, -> { where('id IN (?)', unscoped.select('MAX(id) as id').group(:company_number).map { |application| application.id }) }
```

Comment:
Understood. That actually makes a lot more sense. I got sidetracked by the development data where all the applications were MembershipApplications records with everything being the same for each record except for the state field.